src/Main.elm

```diff
@@ -92,37 +126,43 @@ showResults float =
 
 type Msg
     = UpdateDisplay CalButton
-    | Saveinput CalButton
+    | Saveinput Action
```

Comment:
I hope this felt like a good change. Having `SaveInput` only be able to hold the things that should actually be used by it is a huge win. A value of `SaveInput CalDecimal` never made sense in the first place, right?